Modules/_winapi.c

```diff
@@ -1512,6 +1512,63 @@ _winapi_PeekNamedPipe_impl(PyObject *module, HANDLE handle, int size)
     }
 }
 
+/*[clinic input]
+_winapi.LCMapStringEx
+
+    locale: LPCWSTR
+    flags: DWORD
+    src: LPCWSTR
+
+[clinic start generated code]*/
+
+static PyObject *
+_winapi_LCMapStringEx_impl(PyObject *module, LPCWSTR locale, DWORD flags,
+                           LPCWSTR src)
+/*[clinic end generated code: output=cf4713d80e2b47c9 input=9fe26f95d5ab0001]*/
+{
+    if (flags & LCMAP_SORTHANDLE) {
+        PyErr_SetString(PyExc_ValueError, "LCMAP_SORTHANDLE is not supported");
+        return NULL;
+    }
+    if (flags & LCMAP_HASH) {
+        PyErr_SetString(PyExc_ValueError, "LCMAP_HASH is not supported");
+        return NULL;
+    }
+    if (flags & LCMAP_BYTEREV) {
+        PyErr_SetString(PyExc_ValueError, "LCMAP_BYTEREV is not supported");
+        return NULL;
+    }
+    if (flags & LCMAP_SORTKEY) {
+        PyErr_SetString(PyExc_ValueError, "LCMAP_SORTKEY is not supported");
+        return NULL;
+    }
+
+    int dest_size = LCMapStringEx(locale, flags, src, wcslen(src), NULL, 0,
+                                  NULL, NULL, 0);
+    if (dest_size == 0) {
+        PyErr_SetFromWindowsErr(0);
+        return NULL;
+    }
+
+    wchar_t* dest = PyMem_NEW(wchar_t, dest_size);
+    if (dest == NULL) {
+        return PyErr_NoMemory();
+    }
+
+    int nmapped = LCMapStringEx(locale, flags, src, wcslen(src), dest, dest_size,
+                                NULL, NULL, 0);
+    if (nmapped == 0) {
+        PyErr_SetFromWindowsErr(0);
+        PyMem_DEL(dest);
+        return NULL;
+    }
```

Comment:
I'd prefer to set the exception just before returning:
```suggestion
    if (nmapped == 0) {
        DWORD error = GetLastError();
        PyMem_DEL(dest);
        return PyErr_SetFromWindowsErr(error);
    }
```